packages/shared/src/lib/feed.ts

```diff
@@ -93,11 +93,27 @@ export type PostAnalyticsEventFnOptions = FeedItemPosition & {
   extra?: Record<string, unknown>;
 };
 
+const feedPathWithIdMatcher = /\/feeds\/(?<feedId>[A-z0-9]{9})\/?/;
+
 export function postAnalyticsEvent(
   eventName: string,
   post: Post | ReadHistoryPost | PostBootData,
   opts?: PostAnalyticsEventFnOptions,
 ): PostItemAnalyticsEvent {
+  const extra = {
+    ...opts?.extra,
+  };
+
+  if (typeof window !== 'undefined') {
+    const currentUrl = new URL(window.location.href);
+
+    const feedPathMatch = currentUrl.pathname.match(feedPathWithIdMatcher);
+
+    if (feedPathMatch?.groups?.feedId) {
+      extra.feed_id = feedPathMatch.groups.feedId;
+    }
+  }
+
```

Comment:
We call this function all over with column and other feed related properties and I tried injecting `extra.feed_id` from there but it was starting to get really messy. 

So I opted into above "hack" to just pull the `feed_id` from current browser URL and inject it. Open for other ideas. FYI @idoshamun

Reply:
As an experiment it's fine. I think it requires extra effort of introducing a feed context once we have time

Reply:
Currently I removed slug URLs from everywhere so regular users won't visit them. Only way it can be if someone is to do it manually.